File: trip-kit-booking-ui/src/main/java/com/skedgo/android/bookingclient/OAuth2CallbackHandler.java
package com.skedgo.android.bookingclient;

import android.app.Activity;
import android.net.Uri;

import com.skedgo.android.tripkit.booking.BookingForm;

import rx.Observable;

public interface OAuth2CallbackHandler {
  Observable<BookingForm> handleOAuthURL(Activity activity, Uri uri, String callback);
  Observable<BookingForm> handleRetryURL(Activity activity, Uri uri);
}

File: trip-kit-booking-ui/src/main/java/com/skedgo/tripkit/booking/ui/OAuth2CallbackHandler.kt
package com.skedgo.tripkit.booking.ui

import android.net.Uri
import com.skedgo.tripkit.booking.BookingForm
import io.reactivex.Observable

interface OAuth2CallbackHandler {
    fun handleOAuthURL(
        bookingForm: BookingForm,
        uri: Uri,
        callback: String
    ): Observable<BookingForm>

    fun handleRetryURL(bookingForm: BookingForm, uri: Uri): Observable<BookingForm>
}